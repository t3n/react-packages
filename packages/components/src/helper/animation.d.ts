import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
