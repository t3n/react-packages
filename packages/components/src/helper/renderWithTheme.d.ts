import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
