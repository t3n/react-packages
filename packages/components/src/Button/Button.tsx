import React, { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';
import styled, { css } from 'styled-components';
import {
  lineHeight,
  margin,
  MarginProps,
  space,
  variant as styledVariant,
  width,
  WidthProps,
} from 'styled-system';

import { composeTextStyle, Theme, ThemeProps } from '@t3n/theme';

import Icon from '../Icon';
import Loader from '../Loader';

export type ButtonAsType = 'button' | 'a';
export type ButtonVariant = 'primary' | 'secondary';
export type ButtonColorVariant = 'default' | 'inverse' | 'highlight';
export type ButtonSizeVariant = 'small' | 'regular' | 'big';

// TODO: Fix polymorphic interface

export interface ButtonProps
  extends ButtonHTMLAttributes<any>,
    Omit<AnchorHTMLAttributes<any>, 'type'>,
    MarginProps,
    WidthProps {
  variant?: ButtonVariant;
  color?: ButtonColorVariant;
  size?: ButtonSizeVariant;

  iconLeft?: React.FC<React.SVGProps<SVGSVGElement>>;
  iconRight?: React.FC<React.SVGProps<SVGSVGElement>>;

  loading?: boolean;
  as?: ButtonAsType;
}

const buildColorVariants = (
  variantProp: ButtonVariant,
  type: 'default' | 'hover',
  theme: Theme
) => {
  const allVariants: ButtonColorVariant[] = ['default', 'highlight', 'inverse'];
  const buildConfig: any = {};

  allVariants.forEach((el) => {
    buildConfig[el] = {
      color:
        variantProp === 'primary'
          ? theme.buttonStyles.color[el][type].color
          : type === 'hover'
          ? theme.buttonStyles.color[el].hover.color
          : theme.buttonStyles.color[el][type].bg,
      bg:
        variantProp === 'primary'
          ? theme.buttonStyles.color[el][type].bg
          : type === 'hover'
          ? theme.buttonStyles.color[el].hover.bg
          : 'transparent',
      borderColor:
        variantProp === 'primary'
          ? theme.buttonStyles.color[el][type].borderColor
          : theme.buttonStyles.color[el][type].bg,
    };
  });
  return buildConfig;
};

export const buttonStyles = css`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  ${width};
  ${margin};

  ${({ theme }: ThemeProps) => `border-radius: ${theme.border.radii[1]}`};
  ${({ theme, size }) =>
    space({ px: 3, py: size && size === 'small' ? 1 : 2, theme })}
  ${({ theme, size }: ButtonProps & ThemeProps) =>
    composeTextStyle({
      textStyle: size === 'big' ? 'big' : 'regular',
      theme,
    })};

  ${({ disabled, loading }: ButtonProps) =>
    disabled || loading ? 'cursor: no-drop' : 'cursor: pointer'};

  /* We have to provide a value for every breakpoint because of specificity
     of line-height from textStyle */
  ${lineHeight({
    lineHeight: ['1.25rem', '1.25rem', '1.25rem', '1.25rem'],
  })}
  ${({ theme, variant: variantProp = 'primary' }: ButtonProps & ThemeProps) =>
    styledVariant({
      prop: 'color',
      variants: buildColorVariants(variantProp, 'default', theme),
    })}

  &:hover :not(:disabled), &:focus :not(:disabled) {
    ${({ theme, variant: variantProp = 'primary' }: ButtonProps & ThemeProps) =>
      styledVariant({
        prop: 'color',
        variants: buildColorVariants(variantProp, 'hover', theme),
      })}

    ${Loader} {
      > div {
        background-color: white;
      }
    }

    ${Icon}${Icon} {
      fill: white;
    }
  }

  ${Loader} {
    > div {
      ${({
        theme,
        color: colorProp,
        variant: variantProp = 'primary',
      }: ButtonProps & ThemeProps) =>
        `background-color: ${
          variantProp === 'secondary'
            ? colorProp === 'highlight' || colorProp === 'inverse'
              ? theme.colors.text.inverse
              : theme.colors.text.primary
            : colorProp === 'highlight' || colorProp === 'inverse'
            ? theme.colors.text.primary
            : theme.colors.text.inverse
        }`};
    }
  }

  ${Icon} {
    transition: fill 0.1s ease-in-out;
    ${({
      variant: variantProp,
      color: colorProp,
      theme,
    }: ThemeProps & ButtonProps) =>
      `fill: ${
        variantProp === 'secondary'
          ? colorProp === 'highlight' || colorProp === 'inverse'
            ? theme.colors.text.inverse
            : theme.colors.text.primary
          : colorProp === 'highlight' || colorProp === 'inverse'
          ? theme.colors.text.primary
          : theme.colors.text.inverse
      }`};
  }

  ${({ loading }) =>
    !loading &&
    css`
      &:disabled {
        opacity: 0.6;

        ${({ color, variant: variantProp, theme }: ThemeProps & ButtonProps) =>
          color &&
          color === 'highlight' &&
          variantProp === 'primary' &&
          css`
            color: ${theme.colors.text.highlight};
          `}
      }
    `}
`;

const StyledButton = styled(
  // eslint-disable-next-line react/button-has-type
  ({ loading, ...rest }: ButtonProps) => <button {...rest} />
)<ButtonProps>`
  ${buttonStyles}
`;

const Button: React.FC<ButtonProps> = ({
  children,
  loading,
  iconLeft,
  iconRight,
  size = 'regular',
  color = 'default',
  variant = 'primary',
  href,
  as,
  onClick,
  disabled,
  ...rest
}) => (
  <StyledButton
    href={href}
    as={href ? 'a' : as}
    size={size}
    color={color}
    variant={variant}
    loading={loading}
    disabled={loading || disabled}
    onClick={(e: any) => !loading && onClick && onClick(e)}
    {...rest}
  >
    {loading ? (
      <Loader small my={2} />
    ) : (
      <>
        {iconLeft && (
          <Icon
            component={iconLeft}
            mr={2}
            width={
              size === 'big' ? '1.5rem' : size === 'small' ? '1rem' : '1.25rem'
            }
            height={
              size === 'big' ? '1.5rem' : size === 'small' ? '1rem' : '1.25rem'
            }
          />
        )}
        {children}
        {iconRight && (
          <Icon component={iconRight} ml={2} width="auto" height="1.5rem" />
        )}
      </>
    )}
  </StyledButton>
);

export default Button;
