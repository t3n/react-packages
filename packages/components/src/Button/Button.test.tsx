import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import Button from './Button';

test('Button in default state matches snapshot', () => {
  const { container } = renderWithTheme(<Button>text</Button>, {});

  expect(container.firstChild).toMatchSnapshot();
});

test('Primary Button as "default" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="primary" color="default">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Primary Button as "inverse" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="primary" color="inverse">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Primary Button as "highlight" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="primary" color="highlight">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Secondary Button as "default" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="secondary" color="default">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Secondary Button as "inverse" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="secondary" color="inverse">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Secondary Button as "highlight" color variant matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button variant="secondary" color="highlight">
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Button in loading state matches snapshot', () => {
  const { container } = renderWithTheme(<Button loading>text</Button>, {});

  expect(container.firstChild).toMatchSnapshot();
});

test('Small Button matches snapshot', () => {
  const { container } = renderWithTheme(<Button size="small">text</Button>, {});

  expect(container.firstChild).toMatchSnapshot();
});

test('Small Button in loading state matches snapshot', () => {
  const { container } = renderWithTheme(
    <Button size="small" loading>
      text
    </Button>,
    {},
  );

  expect(container.firstChild).toMatchSnapshot();
});
