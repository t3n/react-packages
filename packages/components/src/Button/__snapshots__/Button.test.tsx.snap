// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button in default state matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  variant="primary"
>
  text
</button>
`;

exports[`Button in loading state matches snapshot 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  margin-top: 8px;
  margin-bottom: 8px;
}

.c1 > div {
  background-color: #f4f4f4;
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
  animation: sk-bouncedelay 1.4s infinite ease-in-out both;
}

.c1 > div:nth-child(1) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c1 > div:nth-child(2) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: no-drop;
  line-height: 1.25rem;
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .c2 > div,
.c0:focus :not(:disabled) .c2 > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .c2 > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  disabled=""
  variant="primary"
>
  <span
    class="c1 c2 "
  >
    <div />
    <div />
    <div />
  </span>
</button>
`;

exports[`Primary Button as "default" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  variant="primary"
>
  text
</button>
`;

exports[`Primary Button as "highlight" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #2a2a2a;
  background-color: #fff;
  border-color: #fff;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #2a2a2a;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #2a2a2a;
}

.c0:disabled {
  opacity: 0.6;
  color: #f9423a;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="highlight"
  variant="primary"
>
  text
</button>
`;

exports[`Primary Button as "inverse" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #2a2a2a;
  background-color: #fff;
  border-color: #fff;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #2a2a2a;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #2a2a2a;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="inverse"
  variant="primary"
>
  text
</button>
`;

exports[`Secondary Button as "default" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #2a2a2a;
  background-color: transparent;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #2a2a2a;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #2a2a2a;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  variant="secondary"
>
  text
</button>
`;

exports[`Secondary Button as "highlight" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #fff;
  background-color: transparent;
  border-color: #fff;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="highlight"
  variant="secondary"
>
  text
</button>
`;

exports[`Secondary Button as "inverse" color variant matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #fff;
  background-color: transparent;
  border-color: #fff;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="inverse"
  variant="secondary"
>
  text
</button>
`;

exports[`Small Button in loading state matches snapshot 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  margin-top: 8px;
  margin-bottom: 8px;
}

.c1 > div {
  background-color: #f4f4f4;
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
  animation: sk-bouncedelay 1.4s infinite ease-in-out both;
}

.c1 > div:nth-child(1) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c1 > div:nth-child(2) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: no-drop;
  line-height: 1.25rem;
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .c2 > div,
.c0:focus :not(:disabled) .c2 > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .c2 > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  disabled=""
  variant="primary"
>
  <span
    class="c1 c2 "
  >
    <div />
    <div />
    <div />
  </span>
</button>
`;

exports[`Small Button matches snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.05s ease-in-out;
  transition: all 0.05s ease-in-out;
  border: 2px solid;
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-family: "adelle-sans",sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  cursor: pointer;
  line-height: 1.25rem;
  color: #fff;
  background-color: #2a2a2a;
  border-color: #2a2a2a;
}

.c0:hover :not(:disabled),
.c0:focus :not(:disabled) {
  color: #fff;
  background-color: #f9423a;
  border-color: #f9423a;
}

.c0:hover :not(:disabled) .sc-AykKD > div,
.c0:focus :not(:disabled) .sc-AykKD > div {
  background-color: white;
}

.c0:hover :not(:disabled) .sc-AykKE.sc-AykKE,
.c0:focus :not(:disabled) .sc-AykKE.sc-AykKE {
  fill: white;
}

.c0 .sc-AykKD > div {
  background-color: #fff;
}

.c0 .sc-AykKE {
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
  fill: #fff;
}

.c0:disabled {
  opacity: 0.6;
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.666em;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    line-height: 1.25rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    line-height: 1.25rem;
  }
}

<button
  class="c0"
  color="default"
  variant="primary"
>
  text
</button>
`;
